DirectProgramming/DPC++FPGA/ReferenceDesigns/gunzip/README.md

```diff
@@ -0,0 +1,281 @@
+   * Compile for emulation (fast compile time, targets emulated FPGA device): 
+     ```
+     nmake fpga_emu
+     ```
+   * Generate the optimization report: 
+     ```
+     nmake report
+     ``` 
+   * An FPGA hardware target is not provided on Windows*. 
+
+*Note:* The Intel® PAC with Intel Arria® 10 GX FPGA and Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX) do not yet support Windows*. Compiling to FPGA hardware on Windows* requires a third-party or custom Board Support Package (BSP) with Windows* support.
+
+### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this Reference Design in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [Intel® oneAPI DPC++ FPGA Workflows on Third-Party IDEs](https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-fpga-workflow-on-ide)
+
+## Running the Reference Design
+
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU).
+     ```
+     ./gunzip.fpga_emu    (Linux)
+     gunzip.fpga_emu.exe  (Windows)
+     ```
+
+2. Run the sample on the FPGA device.
+     ```
+     ./gunzip.fpga        (Linux)
+     ```
+
+### Example of Output
+You should see output similar to the following in the console:
+```
+>>>>> Uncompressed File Test <<<<<
+Decompressing '../data/uncompressed.gz' 1 time
+Launching kernels for run 0
+All kernels have finished for run 0
+>>>>> Uncompressed File Test: PASSED <<<<<
+
+>>>>> Statically Compressed File Test <<<<<
+Decompressing '../data/static_compressed.gz' 1 time
+Launching kernels for run 0
+All kernels have finished for run 0
+>>>>> Statically Compressed File Test: PASSED <<<<<
+
+>>>>> Dynamically Compressed File Test <<<<<
+Decompressing '../data/dynamic_compressed.gz' 1 time
+Launching kernels for run 0
+All kernels have finished for run 0
+>>>>> Dynamically Compressed File Test: PASSED <<<<<
+
+>>>>> Throughput Test <<<<<
+Decompressing '../data/tp_test.gz' 5 times
+Launching kernels for run 0
+All kernels have finished for run 0
+Launching kernels for run 1
+All kernels have finished for run 1
+Launching kernels for run 2
+All kernels have finished for run 2
+Launching kernels for run 3
+All kernels have finished for run 3
+Launching kernels for run 4
+All kernels have finished for run 4
+Execution time: 7.12956 ms
+Output Throughput: 561.045 MB/s
+Compression Ratio: 1021.26
+>>>>> Throughput Test: PASSED <<<<<
+
+PASSED
+```
+
+NOTE: When running on the FPGA emulator, the *Execution time* and *Throughput* do not reflect the design's actual hardware performance. <br>
+NOTE: The throughput of the decompression engine is highly dependant on the compression ratio of the input file, more specifically on the amount of LZ77 compression. The throughput test reports the *maximum* throughput, since it is decompressing a highly compressed file. When decompressing other files, it is expected that the throughput will be lower.
```

Comment:
```suggestion
NOTE: The throughput of the decompression engine is highly dependant on the compression ratio of the input file, more specifically on the amount of LZ77 compression. The file chosen for this test has a very high amount of LZ77 compression, and was chosen to demonstrate the *maximum* throughput achievable with this implementation. When decompressing other files, the throughput may be somewhat lower.
```
Can we put a lower bound on how much lower?  'In the worst case, as much as xx% (50%?) lower'

Reply:
The lowest would be a completely uncompressed file, which would be like 90% decrease.